src/GafferScene/SceneAlgo.cpp

```diff
@@ -475,6 +485,16 @@ SceneAlgo::History::Ptr historyWalk( const CapturedProcess *process, InternedStr
 	// (inclusive of each).
 
 	SceneAlgo::History::Ptr result;
+	if( !process->plug )
+	{
+		// CapturedProcess entries with a null plug are just sentinels for plugs that we aren't interested in
+		// TODO - would it be better for sentinels to not store themselves in parent->children()?
+		// m_processMap doesn't store smart pointers with ownership, but we know that we are going to create them
+		// in processStarted and finish with them in processFinished, so we could just do manual memory management
+		// with new/delete
```

Comment:
Ah, I see why you're storing them now - something has to keep them alive. It does seem wrong for CapturingMonitor to capture things we're not interested in though, and then for us to repeat the filtering code later.

Can we store `variant<CapturedProcess *, MonitorScope>` in the map instead? I think that could be quite a lot cleaner for a couple of reasons :

- The data structure enforces the "either capture, or disable future captures" behaviour we are trying to implement.
- `processFinished()` can go back to being a simple `erase()`.
- We can remove all the manual memory management.
- `historyWalk()` won't have to do any filtering at all (we can remove the previous filtering even).